lib/util/esWrapper.js

```diff
@@ -116,6 +116,12 @@ const errorMessagesMapping = [
     regex: /^\[(.*)\]: version conflict, document already exists.*/,
     subcode: 'document_already_exists',
     getPlaceholders: matches => [matches[1]]
+  },
+  {
+    // Unknown key for a START_OBJECT in [term].
+    regex: /^Unknown key for a START_OBJECT in \[(.*)\].*/,
+    subcode: 'cannot_parse_query',
```

Comment:
Merge warning: this error code doesn't exist anymore. Might be `unexpected_properties` except that it requires a field name. Maybe a new `services.storage.forbidden_argument` following the same model as `api.assert.forbidden_argument`?

Reply:
Yes I agree with `services.storage.forbidden_argument`, I will change that when your PR is merged

Reply:
Finally I create `services.storage.invalid_search_query`, it was more explicit